core/src/main/java/org/apache/calcite/rel/metadata/RelMdExpressionLineage.java

```diff
@@ -391,6 +392,18 @@ protected RelMdExpressionLineage() {}
     return mq.getExpressionLineage(rel.getInput(), outputExpression);
   }
 
+  /**
+   * Expression lineage from Snapshot.
+   * @param rel Relational expression which is Snapshot
+   * @param mq Metadata query
+   * @param outputExpression Expression which need be inferred
+   * @return If the lineage cannot be inferred, we return null.
+   */
```

Comment:
Thanks for adding the javadoc, in general the sentence starts with lower-case letters, I have taken the liberty to also rephrase a couple of sentences. No problem if you want to keep the original sentences, but please fix the first letter casing if you do so.

```suggestion
  /**
   * Expression lineage from Snapshot.
   * @param rel Snapshot relational expression
   * @param mq metadata query
   * @param outputExpression expression which needs to be inferred
   * @return the inferred lineage, possibly null.
   */
```